src/controller/java/src/chip/devicecontroller/model/ClusterState.java

```diff
@@ -51,6 +55,33 @@ public Optional<Long> getDataVersion() {
     return dataVersion;
   }
 
+  /**
+   * Convenience utility for getting all attributes in Json string format.
+   *
+   * @return all attributes in Json string format., or empty string if not found.
+   */
+  public String getAttributesJson() {
+    JSONObject combinedObject = new JSONObject();
+    attributes.forEach(
+        (attributeId, attributeState) -> {
+          JSONObject attributeJsonObj = attributeState.getJson();
+          if (attributeJsonObj != null) {
+            Iterator<String> keys = attributeJsonObj.keys();
+            while (keys.hasNext()) {
+              String key = keys.next();
+              if (combinedObject.has(key)) {
+                Log.e(TAG, "Conflicting attribute tag Id is found: " + key);
+                break;
```

Comment:
this kills the entire loop, so any future keys are ignored. Is this intentional? Why not continue or raise exception if truly a final error?

Reply:
yes, it is intentional. this kind of error should never happen.

Reply:
updated to continue to the end and show all possible conflicting error. thanks